benchmarks/src/jmh/kotlin/com/target/nativememoryallocator/benchmarks/impl/NMAOffHeapCache.kt

```diff
@@ -7,45 +7,46 @@ import com.target.nativememoryallocator.map.NativeMemoryMapBackend
 import com.target.nativememoryallocator.map.NativeMemoryMapBuilder
 import com.target.nativememoryallocator.map.NativeMemoryMapSerializer
 import mu.KotlinLogging
+import java.nio.ByteBuffer
 
 private val logger = KotlinLogging.logger {}
 
 
 /**
  * NMA implementation of OffHeapCache.
  */
-class NMAOffHeapCache : OffHeapCache<String, ByteArray> {
+class NMAOffHeapCache : OffHeapCache<String, ByteBuffer> {
 
     init {
-        logger.info { "initializing NMAOffHeapCache" }
+        logger.info { "initializing NMAOffHeapCache with asByteBuffer change" }
     }
 
     private val nativeMemoryAllocator = NativeMemoryAllocatorBuilder(
         pageSizeBytes = 4_096, //4KB
         nativeMemorySizeBytes = (10L * 1024 * 1024 * 1024), //10gb,
     ).build()
 
-    private val valueSerializer = object : NativeMemoryMapSerializer<ByteArray> {
-        override fun deserializeFromOnHeapMemoryBuffer(onHeapMemoryBuffer: OnHeapMemoryBuffer): ByteArray {
-            return onHeapMemoryBuffer.toTrimmedArray()
+    private val valueSerializer = object : NativeMemoryMapSerializer<ByteBuffer> {
+        override fun deserializeFromOnHeapMemoryBuffer(onHeapMemoryBuffer: OnHeapMemoryBuffer): ByteBuffer {
+            return onHeapMemoryBuffer.asByteBuffer()
```

Comment:
`toTrimmedArray` makes a copy of the byte array on every read, `asByteBuffer` does not make a copy - this takes advantage of the thread-local read buffer in `NativeMemoryMap`